iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/PrivilegedUserCheck.java

```diff
@@ -83,8 +84,11 @@ private void handle(CheckContext ctx, DockerImage dockerImage) {
     if(!isLastDockerImageInFile(dockerImage)) {
       return;
     }
-    String imageName = dockerImage.from().image().name().value();
+    String imageName = getImageName(dockerImage.from());
     Optional<UserInstruction> lastUser = getLastUser(dockerImage);
+    if (imageName == null) {
```

Comment:
This `if` block can be before `getLastUser()` call - it's not necessary to call it when `imageName` is `null`.